### PR TITLE
Requested fix for ruby-based C-A-R + unrelated cosmetic work

### PR DESCRIPTION
**Pull request by http-418**
This PR contains two important fixes:
- Warn the user and bail if cluster.txt is missing critical roles on head nodes
- Add `liblzo2` dependency for Spark 2.x to be able to read gzip files

Additionally there are minor, unrelated fixes:
- Make the yarn `min.user.id` parameterized to make VM clusters more pleasant to use with service principals
- Make sure ignore_failure is disabled on CentOS and Ubuntu 16.04 in `bcpc::ssh`
- Add copyright/license header to `bach_spark::default`

### CODE REVIEW

File: cookbooks/bach_spark/recipes/default.rb [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This recipe installs spark to /usr/spark/ using a package
```
Comment (cbaenziger) [inline cookbooks/bach_spark/recipes/default.rb]: Not just `/usr/spark` but to `node[:spark][:bin][:dir]` right?

**Comment by cbaenziger** [comment]
Being this uses `all_hadoop_head_nodes` does this also protect Kafka clusters too?

File: cookbooks/bach_spark/recipes/default.rb (hunk 1 above)
Comment (http-418) [inline cookbooks/bach_spark/recipes/default.rb]: It's hardcoded in the package, so no.

**Comment by http-418** [comment]
The role is called BCPC-Hadoop-Head on both Hadoop and Kafka clusters.

File: cookbooks/bach_spark/recipes/default.rb (hunk 1 above)
Comment (cbaenziger) [inline cookbooks/bach_spark/recipes/default.rb]: Ouch!

**Pull request merged**